webapp/src/pages/PerformanceAnalysis.test.tsx

```diff
@@ -0,0 +1,36 @@
+import { screen } from "@testing-library/react";
+import { renderWithRouterAndRedux } from "mocks/utils";
+import PerformanceAnalysis from "./PerformanceAnalysis";
+import { setupServer } from "msw/node";
+import { getDatasetInfoAPIResponse } from "mocks/api/mockDatasetInfoAPI";
+import { PIPELINE_REQUIRED_TIP } from "utils/const";
+
+describe("PerformanceAnalysisComparisonWithoutPipeline", () => {
+  const handlers = [getDatasetInfoAPIResponse];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  it("should display 'pipeline requied' message if no pipeline selected or exist in the config", () => {
+    renderWithRouterAndRedux(<PerformanceAnalysis />, {
+      route: "/local/dataset_splits/eval/pipeline_metrics",
+      path: "/:jobId/dataset_splits/:datasetSplitName/pipeline_metrics",
+    });
+    expect(screen.getByText(PIPELINE_REQUIRED_TIP)).toBeVisible();
+    expect(
+      screen.getByText("Pipeline Metrics by Data Subpopulation")
+    ).toBeVisible();
+    expect(
+      screen.getByText("Analyze metrics for different data subpopulations.")
+    ).toBeVisible();
+    const link: HTMLAnchorElement = screen.getByRole("link");
+    expect(link).toHaveAttribute(
```

Comment:
Is the type necessary here? I would simply
```suggestion
    expect(screen.getByRole("link")).toHaveAttribute(
```

Reply:
Sure.